Mathlib/LinearAlgebra/QuadraticForm/TensorProduct.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.LinearAlgebra.BilinearForm.TensorProduct
+import Mathlib.LinearAlgebra.QuadraticForm.Basic
+
+/-!
+# The quadratic form on a tensor product
+
+## Main definitions
+
+* `QuadraticForm.tensorDistrib (Q₁ ⊗ₜ Q₂)`: the quadratic form on `M₁ ⊗ M₂` constructed by applying
+  `Q₁` on `M₁` and `Q₂` on `M₂`. This construction is not available in characteristic two.
+
+-/
+
+universe uι uR uA uM₁ uM₂
+
+variable {ι : Type uι} {R : Type uR} {A : Type uA} {M₁ : Type uM₁} {M₂ : Type uM₂}
+
+open TensorProduct
+
+namespace QuadraticForm
+
+section CommRing
+variable [CommRing R] [CommRing A]
+variable [AddCommGroup M₁] [AddCommGroup M₂]
+variable [Algebra R A] [Module R M₁] [Module A M₁]
+variable [SMulCommClass R A M₁] [SMulCommClass A R M₁] [IsScalarTower R A M₁]
+variable [Module R M₂] [Invertible (2 : R)]
+
+
+variable (R A) in
+/-- The tensor product of two quadratic forms injects into quadratic forms on tensor products.
+
+Note this is heterobasic; the quadratic form on the left can take values in a larger ring than
+the one on the right. -/
+def tensorDistrib : QuadraticForm A M₁ ⊗[R] QuadraticForm R M₂ →ₗ[A] QuadraticForm A (M₁ ⊗[R] M₂) :=
+  letI : Invertible (2 : A) := (Invertible.map (algebraMap R A) 2).copy 2 (map_ofNat _ _).symm
+  let toQ := BilinForm.toQuadraticFormLinearMap A A (M₁ ⊗[R] M₂)
```

Comment:
Would these have (marginally) cleaner definitional properties as `letI`'s instead?

Reply:
These lets are indeed here as an attempt to fight timeouts

Reply:
Looks like `let` is 1% more heartbeats but 33% more wall time, so I'd prefer to leave this as is.

Reply:
> These lets are indeed here as an attempt to fight timeouts

I suggest adding a comment / remark documenting that performance has impacted style here.

Reply:
```suggestion
  -- while `letI`s would produce a better term than `let`, they would make this already-slow
  -- definition even slower.
  let toQ := BilinForm.toQuadraticFormLinearMap A A (M₁ ⊗[R] M₂)
```